test/release/examples/primers/interopWithC.chpl

```diff
@@ -0,0 +1,332 @@
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 8-9
+*/
+
+//
+// After that, the exported functions may be called:
+//
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 11-13
+*/
+
+/*
+   When ending the C program, the user must explicitly shut down the Chapel
+   runtime and module code.  This is done by calling ``chpl_library_finalize``:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 15-19
+*/
+
+/*
+   Once shut down, the Chapel runtime cannot be restarted in the current C
+   program.
+*/
+
+/*
+   .. _primers-C-interop-compiling-2:
+
+   Compiling C Code with the Chapel Library
+   ++++++++++++++++++++++++++++++++++++++++
+*/
+
+/*
+   Compiling C code with the generated Chapel library file is generally complex,
+   but can be made simpler through the use of the ``--library-makefile`` flag
+   as described in :ref:`readme-libraries-linking`.
+   
+   An example of compiling a C program with a generated Chapel library using the
+   generated Makefile can be found in the `Makefile
+   <https://github.com/chapel-lang/chapel/blob/master/test/release/examples/primers/Makefile>`_
+   for the primers directory, to build this source file.
+*/ 
+
+/*
+   .. _primers-C-interop-using-C:
+
+   Using C Code in Chapel
+   ----------------------
+*/
+
+/*
+   Chapel has support for C code but we need to tell the compiler
+   about the C symbols using the ``extern`` keyword
+*/
+
+   extern proc baz(): int;
+
+//
+// The function can then be called as normal:
+//
+   proc callsBaz() {
+      writeln(baz());
+   }
+
+//
+// Such functions can even be called in exported functions:
+//
+
+   export proc alsoCallsBaz() {
+      writeln(baz());
+      callsBaz();
+   }
+
+// You can tell the chapel compiler where to look for these C functions by adding a
+// require statement
+   require "cHelper.h", "cHelper.c";
+// You must inlcude both a header file and C file.
+// Chapel also supports require statements for ``.o`` files and
+// for archived libraries using the ``-l`` flag
+/*
+   .. code-block:: chapel
+
+         require "foo.o", "-lapache-arrow";
+*/
+
+// Alternatively you can also include their names while invoking the chapel compiler.
+
+//
+// Unlike ``export``, ``extern`` can also be applied to global variables,
+// struct definitions, or even typedefs.
+//
+   extern var x: int(32);
+   extern var y: uint(32) = 3;
+   extern type myType = int;
+   extern proc useMyType(arg: myType): int; // an extern function using the typedef
+
+   export proc callUseMyType() {
+      var blah: myType = 17;
+      writeln(useMyType(blah));
+   }
+
+// Chapel also has a standard module named :mod:`SysCTypes` (located under
+//``$CHPL_HOME/modules/standard/gen/...``) which defines a few C types which
```

Comment:
This sentence is a bit run-on.  I'd maybe split it in two at the first "which", starting the second sentence with "This module defines..."